### PR TITLE
DM-18836: Add setup.py to ts_salobj

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    packages=find_namespace_packages(where="python"),
+    package_data={"": ["*.rst", "*.yaml"]},
+    scripts=["bin.src/purge_topics.py", "bin.src/run_test_csc.py", "bin.src/show_salpy_attributes.py"],
+    test_requires=["pytest", "pytest-cov", "pytest-flake8"],
+    license="GPL",
+    project_urls={
+        "Bug Tracker": "https://jira.lsstcorp.org/secure/Dashboard.jspa",
+        "Source Code": "https://github.com/lsst-ts/ts_salobj",
+    }
+)
```
Comment (couger01) [inline setup.py]:
```
from setuptools import setup, find_namespace_packages
install_requires=["numpy"]
tests_require=["pytest", "pytest-cov", "pytest-flake8"]
setup(
    name="ts_salobj",
    description="Object-oriented interface to LSST SAL communications middleware.",
    use_scm_version={"write_to":"python/lsst/ts/salobj/version.py"},
    setup_requires=["setuptools-scm", "pytest-runner"],
    install_requires=install_requires,
    package_dir={"": "python"},
    packages=find_namespace_packages(where="python"),
    package_data={"": ["*.rst", "*.yaml"]},
    scripts=["bin.src/purge_topics.py", "bin.src/run_test_csc.py", "bin.src/show_salpy_attributes.py"],
    tests_require=tests_require,
    extras_require={"dev": install_requires + tests_require + ["documenteer[pipelines]"]},
    license="GPL",
    project_urls={
        "Bug Tracker": "https://jira.lsstcorp.org/secure/Dashboard.jspa",
        "Source Code": "https://github.com/lsst-ts/ts_salobj",
    }
)
```

This is what worked for me. I added an extras_require that allows the user to specify the dev dependencies. I also corrected the test_requires parameter to be tests_require parameter. Also numpy is an installation requirement. There is a version.py generated and so I have setuptools_scm write a version attribute to that module.

Comment (couger01) [inline setup.py]:
```
"""Sphinx configuration file for an LSST stack package.

This configuration only affects single-package Sphinx documentation builds.
"""

from documenteer.sphinxconfig.stackconf import build_package_configs
import lsst.ts.salobj


_g = globals()
_g.update(build_package_configs(
    project_name='ts_salobj',
    version=lsst.ts.salobj.version))
```

I also changed the version to reflect the version.py changes.

Comment (r-owen) [reply setup.py]: Shouldn't this be `version=lsst.ts.salobj.version.__version__`?

Comment (r-owen) [reply setup.py]: OK. Pity about the confusing x_requires vs ys_require keywords but I see it's the standard.

**Comment by r-owen** [comment]
I made the requested changes except for the breaking change of having `setuptools_scm` write `version` to `version.py` file and the associated change to the sphinx config file. The Python standard is `__version__` not `version` and scons and sphinx presently use that standard.

**Comment by r-owen** [comment]
We figured out a way to write `__version__` in `version.py` so I think it's now complete.

**Pull request merged**